dmriprep/interfaces/vectors.py

```diff
@@ -1,91 +1,101 @@
+"""Handling the gradient table."""
+from pathlib import Path
+import numpy as np
+from nipype.utils.filemanip import fname_presuffix
+from nipype.interfaces.base import (
+    SimpleInterface, BaseInterfaceInputSpec, TraitedSpec,
+    File, traits, isdefined
+)
+from ..utils.vectors import DiffusionGradientTable, B0_THRESHOLD, BVEC_NORM_EPSILON
+
+
+class _CheckGradientTableInputSpec(BaseInterfaceInputSpec):
     dwi_file = File(exists=True, mandatory=True)
-    fbval = File(exists=False, mandatory=False)
-    fbvec = File(exists=False, mandatory=False)
-    rasb_tsv_file = File(exists=False, mandatory=False)
-    rescale = traits.Bool(True, usedefault=True)
-    save_fsl_style = traits.Bool(True, usedefault=True)
-    B0_NORM_EPSILON = traits.Float(50)
-    bval_scaling = traits.Bool(True, usedefault=True)
-
-
-class CleanVecsOutputSpec(TraitedSpec):
-    """Output interface wrapper for CleanVecs"""
-    from dipy.core.gradients import GradientTable
-    gtab = traits.Instance(GradientTable)
-    is_hemi = traits.Bool()
-    pole = traits.Array()
-    b0_ixs = traits.Array()
-    slm = traits.Str()
-    fbval_out_file = traits.Any()
-    fbvec_out_file = traits.Any()
-    rasb_tsv_out_file = File(exists=True, mandatory=True)
-
-
-class CleanVecs(SimpleInterface):
-    """Interface wrapper for CleanVecs"""
-    input_spec = CleanVecsInputSpec
-    output_spec = CleanVecsOutputSpec
+    in_bvec = File(exists=True, xor=['in_rasb'])
+    in_bval = File(exists=True, xor=['in_rasb'])
+    in_rasb = File(exists=True, xor=['in_bval', 'in_bvec'])
+    b0_threshold = traits.Float(B0_THRESHOLD, usedefault=True)
+    bvec_norm_epsilon = traits.Float(BVEC_NORM_EPSILON, usedefault=True)
+    b_scale = traits.Bool(True, usedefault=True)
+
+
+class _CheckGradientTableOutputSpec(TraitedSpec):
+    out_rasb = File(exists=True)
+    out_bval = File(exists=True)
+    out_bvec = File(exists=True)
+    full_sphere = traits.Bool()
+    pole = traits.Tuple(traits.Float, traits.Float, traits.Float)
+    b0_ixs = traits.List(traits.Int)
+
+
+class CheckGradientTable(SimpleInterface):
+    """
+    Ensure the correctness of the gradient table.
+
+    Example
+    -------
+
+    >>> os.chdir(tmpdir)
+    >>> check = CheckGradientTable(
+    ...     dwi_file=str(data_dir / 'dwi.nii.gz'),
+    ...     in_rasb=str(data_dir / 'dwi.tsv')).run()
+    >>> check.outputs.pole
+    (0.0, 0.0, 0.0)
+    >>> check.outputs.full_sphere
+    True
+
+    >>> check = CheckGradientTable(
+    ...     dwi_file=str(data_dir / 'dwi.nii.gz'),
+    ...     in_bvec=str(data_dir / 'bvec'),
+    ...     in_bval=str(data_dir / 'bval')).run()
+    >>> check.outputs.pole
+    (0.0, 0.0, 0.0)
+    >>> check.outputs.full_sphere
+    True
+    >>> newrasb = np.loadtxt(check.outputs.out_rasb, skiprows=1)
+    >>> oldrasb = np.loadtxt(str(data_dir / 'dwi.tsv'), skiprows=1)
+    >>> np.allclose(newrasb, oldrasb, rtol=1.e-3)
+    True
+
+    """
+
+    input_spec = _CheckGradientTableInputSpec
```

Comment:
Underscores hide these classes from the name resolvers and you can also leave the docstring unset for them.